frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -0,0 +1,485 @@
+      type: BREADCRUMBS_TYPE.APP_FULLSCREEN
+    }]
+    if (state.folderId) {
+      breadcrumbsList.push({
+        link: <Link to={`/ui/workspaces/${state.config.appConfig.workspaceId}/gallery?folder_ids=${state.folderId}`}>{state.fileName}</Link>,
+        type: BREADCRUMBS_TYPE.APP_FULLSCREEN
+      })
+    }
+    breadcrumbsList.push({
+      link: <Link to={`/ui/workspaces/${state.config.appConfig.workspaceId}/contents/file/${state.imagesPreviews[state.fileSelected].contentId}`}>{state.imagesPreviews[state.fileSelected].fileName}</Link>,
+      type: BREADCRUMBS_TYPE.APP_FULLSCREEN
+    })
+
+    // FIXME - CH - 2019/04/25 - We should keep redux breadcrumbs sync with fullscreen apps but when do the setBreadcrumbs,
+    // app crash telling it cannot render a Link outside a router
+    // see https://github.com/tracim/tracim/issues/1637
+    // GLOBAL_dispatchEvent({type: 'setBreadcrumbs', data: {breadcrumbs: breadcrumbsList}})
+    this.setState({ breadcrumbsList: breadcrumbsList })
+  }
+
+  loadFolderDetail = async (workspaceId, folderId) => {
+    const { state, props } = this
+
+    const fetchContentDetail = await handleFetchResult(
+      await getFolderDetail(state.config.apiUrl, workspaceId, folderId)
+    )
+
+    switch (fetchContentDetail.apiResponse.status) {
+      case 200:
+        this.setState({ fileName: fetchContentDetail.body.filename })
+        break
+      default: this.sendGlobalFlashMessage(props.t('Error while loading folder detail'))
+    }
+  }
+
+  loadGalleryList = async (workspaceId, folderId) => {
+    const { state, props } = this
+
+    let fetchContentList
+    if (folderId) {
+      fetchContentList = await handleFetchResult(
+        await getFolderContentList(state.config.apiUrl, workspaceId, folderId)
+      )
+    } else {
+      fetchContentList = await handleFetchResult(
+        await getWorkspaceContentList(state.config.apiUrl, workspaceId)
+      )
+    }
+
+    switch (fetchContentList.apiResponse.status) {
+      case 200:
+        let images = []
+        fetchContentList.body.forEach(content => {
+          if (content.content_type === 'file') {
+            images.push({ src: '', contentId: content.content_id })
+          }
+        })
+        const imagesPreviews = await this.loadPreview(images)
+
+        this.setState({ imagesPreviews })
+
+        break
+      default: this.sendGlobalFlashMessage(props.t('Error while loading content list'))
+    }
+  }
+
+  loadPreview = async (images) => {
+    const { state, props } = this
+
+    return Promise.all(images.map(async (image) => {
+      const fetchFileContent = await handleFetchResult(
+        await getFileContent(state.config.apiUrl, state.config.appConfig.workspaceId, image.contentId)
+      )
+      switch (fetchFileContent.apiResponse.status) {
+        case 200:
+          const filenameNoExtension = removeExtensionOfFilename(fetchFileContent.body.filename)
```

Comment:
removeExtensionOfFilename is a duplicate from app File, don't duplicate. Instead, add it to frontend_lib